darwin/datatypes.py

```diff
@@ -86,43 +87,44 @@ def full_path(self) -> str:
         return construct_full_path(self.remote_path, self.filename)
 
 
-def make_bounding_box(class_name, x, y, w, h):
+def make_bounding_box(class_name: str, x: float, y: float, w: float, h: float, subs: List[SubAnnotation] = []):
     return Annotation(
         AnnotationClass(class_name, "bounding_box"),
         {"x": round(x, 3), "y": round(y, 3), "w": round(w, 3), "h": round(h, 3)},
+        subs,
     )
 
 
-def make_tag(class_name):
-    return Annotation(AnnotationClass(class_name, "tag"), {})
+def make_tag(class_name: str, subs: List[SubAnnotation] = []):
+    return Annotation(AnnotationClass(class_name, "tag"), {}, subs)
 
 
-def make_polygon(class_name, point_path):
-    return Annotation(AnnotationClass(class_name, "polygon"), {"path": point_path})
+def make_polygon(class_name: str, point_path: List[Point], subs: List[SubAnnotation] = []):
+    return Annotation(AnnotationClass(class_name, "polygon"), {"path": point_path}, subs)
 
 
-def make_complex_polygon(class_name, point_paths):
-    return Annotation(AnnotationClass(class_name, "complex_polygon", "polygon"), {"paths": point_paths})
+def make_complex_polygon(class_name: str, point_paths: List[List[Point]], subs: List[SubAnnotation] = []):
+    return Annotation(AnnotationClass(class_name, "complex_polygon", "polygon"), {"paths": point_paths}, subs)
 
 
-def make_keypoint(class_name, x, y):
-    return Annotation(AnnotationClass(class_name, "keypoint"), {"x": x, "y": y})
+def make_keypoint(class_name: str, x: float, y: float, subs: List[SubAnnotation] = []):
+    return Annotation(AnnotationClass(class_name, "keypoint"), {"x": x, "y": y}, subs)
 
 
-def make_line(class_name, path):
-    return Annotation(AnnotationClass(class_name, "line"), {"path": path})
+def make_line(class_name: str, path: List[Point], subs: List[SubAnnotation] = []):
```

Comment:
Please remove the default parameter `=[]`. Default list (/dict/tuple) in parameters are evaluate ad function definition time, so they points always to the same instance creating weird side effects:

```python
>>> def foo(a=[]):
...     a.append('a')
...     return a
... 
>>> foo()
['a']
>>> foo()
['a', 'a']
>>> foo()
['a', 'a', 'a']
```
We can make them `Optional` and `=None`

Reply:
Great point, will change that